packages/cli/yaml/docs-validator/src/rules/valid-markdown-link/collect-pathnames.ts

```diff
@@ -71,7 +90,12 @@ export function collectPathnamesToCheck(
 
         pathnamesToCheck.push({
             pathname,
-            position: link.position,
+            sourceFilepath: link.sourceFilepath,
+            position: link.position && {
+                ...link.position,
+                start: { ...link.position.start, line: link.position.start.line + frontmatterOffset },
+                end: { ...link.position.end, line: link.position.end.line + frontmatterOffset }
+            },
```

Comment:
nit: seems like we have the same logic copy pasta-ed in a couple places

```ts
sourceFilepath: link.sourceFilepath,
            position: link.position && {
                ...link.position,
                start: { ...link.position.start, line: link.position.start.line + frontmatterOffset },
                end: { ...link.position.end, line: link.position.end.line + frontmatterOffset }
            },
```

i could see abstracting that to a single helper